apps/arweave/src/ar_node_state.erl

```diff
@@ -38,7 +38,8 @@ start() ->
 		{cumulative_diff, 0},               % Sum of the difficulty squared along the current weave
 		{hash_list_merkle, <<>>},           % The Merkle root of the current hash list
 		{block_txs_pairs, []},              % List of {BH, TXIDs} pairs for last ?MAX_TX_ANCHOR_DEPTH blocks
-		{mempool_size, 0}                   % Memory pool size
+		{mempool_size, 0},                  % Memory pool size
+		{txs_pending_download, sets:new()}  % set of txs pending download (only TXIDs)
```

Comment:
Not sure if I understand the purpose of this set. It should be sufficient to go from the most recent block to the oldest and download the missing transactions from them, we do not really need them until they are mined.

Reply:
ok, purpose is for controlling & determining whether the downloader can be scheduled for execution here https://github.com/ArweaveTeam/arweave/pull/216/commits/72e1d2f38cdc460137431ddc9916a347adf43413#diff-47fb75d8dee1a57e1c5e1a6dfae18c09R149-R150 or not - with its disk activity posing as being potentially "expensive", and with only format-2 TX (IDs) qualifying to be in this set. If empty, we don't even attempt to interact & trigger the downloader https://github.com/ArweaveTeam/arweave/pull/216/commits/72e1d2f38cdc460137431ddc9916a347adf43413#diff-47fb75d8dee1a57e1c5e1a6dfae18c09R738